### ISSUE TITLE
Enigma: Handle nested PGP encryption

### ISSUE THREAD

**Issue by aacebedo** [roundcube/roundcubemail#5634]
My mail provider offers a way to encrypt incoming emails with PGP even if the sender didn't used PGP to send the email. In this case Roundcube is able to decrypt the mail as it thinks the mail have been encrypted by the sender.
However when the sender really uses PGP encryption, the mail server re-encrypt a second time the incoming mail resulting in a nested PGP encryption.
Thunderbird with enigmail is able to handle this nested encryption and transparently decrypt the incoming mail.
Is it possible to implement this feature on the enigma plugin ?

I am using roundcube 1.3 beta

**Comment by alecpl** [comment]
I'd like to know what is the message format (of both messages), plain or mime pgp?

**Comment by aacebedo** [comment]
Hi

I am not entirely sure when to talk about plain or mime but here is what I've found based on analyzing the messages.

First level of encryption (sender side) cannot be guaranteed as the sender's application may change the format according to settings. However I have tested with Mailvelope (https://www.mailvelope.com/en) and it sends it as plain text.

Second level of encryption (made by the mail provider) uses pgp mime (Content-type="multipart/encrypted" with protocol="application/pgp-encrypted"
